test/Microsoft.AspNet.Mvc.IntegrationTests/BinderTypeBasedModelBinderIntegrationTest.cs

```diff
@@ -0,0 +1,305 @@
+            Assert.Null(modelBindingResult.Model);
+            Assert.Equal(isModelSet, modelBindingResult.IsModelSet);
+
+            // ModelState
+            Assert.True(modelState.IsValid);
+            var key = Assert.Single(modelState.Keys);
+            Assert.Equal("CustomParameter", key);
+            Assert.Empty(modelState[key].Errors);
+            Assert.Equal(ModelValidationState.Valid, modelState[key].ValidationState);
+            Assert.Null(modelState[key].Value); // value is only set if the custom model binder sets it.
+        }
+
+        private class Person2
+        {
+        }
+
+        [Fact]
+        public async Task BindParameter_WithModelBinderType_NonGreedy_NoData()
+        {
+            // Arrange
+            var argumentBinder = ModelBindingTestHelper.GetArgumentBinder();
+            var parameter = new ParameterDescriptor()
+            {
+                Name = "Parameter1",
+                BindingInfo = new BindingInfo()
+                {
+                    BinderType = typeof(NullResultModelBinder)
+                },
+
+                ParameterType = typeof(Person2)
+            };
+
+            // No data is passed.
+            var operationContext = ModelBindingTestHelper.GetOperationBindingContext(request => { });
+            var modelState = new ModelStateDictionary();
+
+            // Act
+            var modelBindingResult = await argumentBinder.BindModelAsync(parameter, modelState, operationContext);
+
+            // Assert
+
+            // ModelBindingResult
+            Assert.Null(modelBindingResult);
+
+            // ModelState
+            Assert.True(modelState.IsValid);
+            Assert.Empty(modelState.Keys);
+        }
+
+        [Fact]
+        public async Task BindParameter_WithOutModelBinderType_NoData()
+        {
+            // Arrange
+            var argumentBinder = ModelBindingTestHelper.GetArgumentBinder();
+            var parameter = new ParameterDescriptor()
+            {
+                Name = "Parameter1",
+                BindingInfo = new BindingInfo()
+                {
+                    BinderType = typeof(NullResultModelBinder)
+                },
+
+                ParameterType = typeof(Person2)
+            };
+
+            // No data is passed.
+            var operationContext = ModelBindingTestHelper.GetOperationBindingContext(request => { });
+            var modelState = new ModelStateDictionary();
+
+            // Act
+            var modelBindingResult = await argumentBinder.BindModelAsync(parameter, modelState, operationContext);
+
+            // Assert
+
+            // ModelBindingResult
+            Assert.Null(modelBindingResult);
+
+            // ModelState
+            Assert.True(modelState.IsValid);
+            Assert.Empty(modelState.Keys);
+        }
+
+        // Ensures that prefix is part of the result returned back.
+        [Fact]
+        public async Task BindParameter_WithData_WithPrefix_GetsBound()
```

Comment:
I get that you're using the convention `BindParameter` -> top level model - `BindProperty` -> not top level model, but you want to change it up because it could be mistaken for action parameter vs controller property.